src/interface.jl

```diff
@@ -97,12 +97,32 @@ isrevertible(transform::Transform) =
   isrevertible(typeof(transform))
 isrevertible(::Type{<:Transform}) = false
 
+# revert does not need to be defined for non-revertible transforms
+function revert(transform::Transform, newobject, cache)
+  if !isrevertible(transform)
+    throw(ArgumentError("Can't revert the non-revertible transform $transform"))
+  end
+  return revert(transform, newobject, cache)
+end
+
 isinvertible(transform::Transform) =
   isinvertible(typeof(transform))
 isinvertible(::Type{<:Transform}) = false
 
+# Base.inv does not need to be defined for non-invertible transforms
+function Base.inv(transform::Transform)
+  if !isinvertible(transform)
+    throw(ArgumentError("Can't invert the non-invertible transform $transform"))
+  end
+  return inv(transform)
+end
+
 preprocess(transform::Transform, object) = nothing
 
+# reapply falls back to apply if not defined
+reapply(transform::Transform, object, cache) =
+  apply(transform, object) 
```

Comment:
Isn't this the situation with StatelessTransform already? Maybe you should subtype from StatelessTransform instead? Maybe we should move this trait to TransformsBase.jl?

Reply:
We  are already in `TransformsBase.jl`, right? Meanwhile, it's true that `StatelessTransform` in `TableTransforms` achieves a similar purpose. The difference is that this lets a transform defaultly behave as a `StatelessTransform` until a `reapply` method is defined for it.

As a side note, I find the interface here easier to implement than its specialization in `TableTransforms`. Not to mention that `TableTransforms` is so big compared to `TransformsBase` for an external package.